### PR TITLE
Polyline new methods

### PR DESCRIPTION
**Pull request by nmaslarinos**
Added a couple of new methods to polyline. 
`polyline.parameter_at` takes as input a Point on the polyline and returns the equivalent parameter. 
`polyline.divide_at_corners` behaves similarly to `split_at_corners`, but instead of returning a list of Polylines, it returns a list of Points that justify the condition. In a way, it completes the duo of functions like divide/split and divide/split_by_length. See the image for the before and after calculations.

Also fixed a bug in `polyline.split_at_corners` where the angle between two consecutive segments was sometimes wrongly calculated.  See the image for the before and after calculations.
Artboard 1-100

### What type of change is this?

- [x] Bug fix in a **backwards-compatible** manner.
- [x] New feature in a **backwards-compatible** manner.

### Checklist

- [x] I added a line to the `CHANGELOG.md` file in the `Unreleased` section under the most fitting heading (e.g. `Added`, `Changed`, `Removed`).
- [x] I ran all tests on my computer and it's all green (i.e. `invoke test`).
- [x] I ran lint on my computer and there are no errors (i.e. `invoke lint`).
- [ ] I added new functions/classes and made them available on a second-level import, e.g. `compas.datastructures.Mesh`.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added necessary documentation (if appropriate)

### CODE REVIEW

**Comment by nmaslarinos** [comment]
As a side-note, I was thinking that a lot of the polyline functions can also be used in the polygon, especially since now there is a method to convert between the two!

File: src/compas/geometry/curves/polyline.py [hunk 1]
```diff
@@ -6,11 +6,11 @@
 
 from compas.geometry import allclose
 from compas.geometry import transform_points
-from compas.geometry import is_point_on_line
+from compas.geometry import is_point_on_line, is_point_on_polyline
 from compas.geometry import Point
 from compas.geometry import Line
 from compas.geometry import Frame
-from .curve import Curve
+from compas.geometry.curves.curve import Curve
```
Comment (tomvanmele) [inline src/compas/geometry/curves/polyline.py]: why change this?

File: tests/compas/geometry/test_curves_polyline.py [hunk 2]
```diff
@@ -431,6 +449,16 @@ def test_polyline_tangent_at_point(coords, input, expected):
     assert expected == Polyline(coords).tangent_at_point(input)
 
 
+@pytest.mark.parametrize("input, expected", [((0, 0, 0), 0.0), ((1, 0, 0), 0.5), ((1, 1, 0), 1.0), ((2, 0, 0), None)])
```
Comment (tomvanmele) [inline tests/compas/geometry/test_curves_polyline.py]: i think the space behind the comma is something the ipy will not like...

Comment (nmaslarinos) [reply tests/compas/geometry/test_curves_polyline.py]: Oops that's my bad, I'm sorry

File: src/compas/geometry/curves/polyline.py (hunk 1 above)
Comment (nmaslarinos) [reply src/compas/geometry/curves/polyline.py]: I think that at the time I was getting an import error, now that I tried again everything works well. I probably forgot to change it to the original when I was done

**Review by tomvanmele — APPROVED** [review]

**Pull request merged**